src/Mvc/test/Mvc.IntegrationTests/ActionParametersIntegrationTest.cs

```diff
@@ -458,8 +521,156 @@ public async Task ActionParameter_BindingToTypeWithNoParameterlessConstructor_Th
             Assert.Equal(
                 string.Format(
                     "Could not create an instance of type '{0}'. Model bound complex types must not be abstract or " +
-                    "value types and must have a parameterless constructor.",
-                    parameterType.FullName),
+                    "value types and must have a parameterless constructor. Record types must have a single primary constructor.",
+                    typeof(AbstractClassWithNoDefaultConstructor).FullName),
+                exception.Message);
+        }
+
+        public class ActionParameter_MultipleConstructors_ParameterlessConstructorModel
+        {
+            public ActionParameter_MultipleConstructors_ParameterlessConstructorModel(string name) => (Name) = name;
+
+            public ActionParameter_MultipleConstructors_ParameterlessConstructorModel(string name, int age) => (Name, Age) = (name, age);
+
+            public ActionParameter_MultipleConstructors_ParameterlessConstructorModel() { }
+
+            public string Name { get; set; }
+
+            public int Age { get; set; }
+        }
+
+        [Fact]
+        public async Task ActionParameter_MultipleConstructors_ParameterlessConstructor()
```

Comment:
Isn't this checked a billion other ways❔ That is, this isn't checking a `record` type and it's unclear why the test was added.